### PR TITLE
Adds rhel-ppc64le to GHA integration tests (merging master changes)

### PR DESCRIPTION
**Pull request by ashwinik30**
In reference with https://github.com/stackrox/collector/pull/1263 and https://github.com/stackrox/collector/pull/1280
Added supoort for ppc64le vsi creation dynamically 
GHA support.
Integration tests on newly created vsi and destroy them dynamically 

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Updated documentation accordingly

**Automated testing**
  - [ ] Added unit tests
  - [ ] Added integration tests
  - [ ] Added regression tests

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why you did not do so. (Valid reasons include "CI is sufficient" or "No testable changes")
In addition to reviewing your code, reviewers **must** also review your testing instructions, and make sure they are sufficient.

For more details, ref the [Confluence page](https://stack-rox.atlassian.net/wiki/spaces/StackRox/pages/855998488/Proposal+Explicitly+List+Testing+Steps+on+PRs) about this section.

### CODE REVIEW

File: ansible/roles/provision-vm/tasks/redhat.yml
Comment (Stringy) [inline ansible/roles/provision-vm/tasks/redhat.yml:27]: This is duplicated with the same block above - we can set the block to run on both s390x and ppc64le to avoid this duplication

File: .github/workflows/integration-tests-vm-type.yml
Comment (Stringy) [inline .github/workflows/integration-tests-vm-type.yml:137]: I need to have a think about how we can appropriately handle this. I think having it in its own step within the workflow is most appropriate, or we can move it into the ansible playbooks, but it should probably move from where it is, and requires some explanation for why it is necessary.

File: ansible/README.md
Comment (Stringy) [inline ansible/README.md:239]:
```suggestion
Note that, for IBM VMs, the destroy process will only delete the VM and release its floating IP. Other shared resources mentioned in the creation process above will not be removed.
```

File: ansible/ansible.cfg
Comment (Stringy) [inline ansible/ansible.cfg:10]: should be removed

File: ansible/ansible.cfg
Comment (Stringy) [inline ansible/ansible.cfg:11]: what is this variable for?

File: ansible/group_vars/all.yml
Comment (Stringy) [inline ansible/group_vars/all.yml:7]: I think we should keep these empty lines

File: ansible/roles/destroy-vm/tasks/main.yml
Comment (Stringy) [inline ansible/roles/destroy-vm/tasks/main.yml:20]: If we can avoid getting the hostname of the VM, these tasks can (and should) be run on localhost - so we can do:

```
- name: Delete IBM Power VSI
  include_tasks: 
      file: destroy-ppc64le-vm.yml
      apply:
        delegate_to: localhost
  when: arch_info == "ppc64le"
```

which will mean we can remove all the lines in `destroy-ppc64le-vm.yml` that delegate to localhost, and just enforce it at a higher level

Comment (Stringy) [inline ansible/roles/destroy-vm/tasks/main.yml:20]: What is the difference between destroying a Power VM and destroying a Z VM? Do we need a separate task file for each one?

**Review by Stringy — CHANGES_REQUESTED** [review]

File: ansible/group_vars/all.yml
Comment (ashwinik30) [reply ansible/group_vars/all.yml:41]: In IBM cloud we have max 56 chars for name 
and currently from CI we are getting more than 56 characters  collector-ci-ebpf-p-6010534240rhelppc64lecpaas
we can only pass here onle one char to match up this

File: ansible/README.md
Comment (ashwinik30) [reply ansible/README.md:239]: noted

File: ansible/ansible.cfg
Comment (ashwinik30) [reply ansible/ansible.cfg:10]: noted

File: ansible/ansible.cfg
Comment (ashwinik30) [reply ansible/ansible.cfg:11]: for the SSH issue debugging purpose this flag was suggested from askubuntu.com 
lets see if that works without this flag

File: ansible/group_vars/all.yml
Comment (ashwinik30) [reply ansible/group_vars/all.yml:7]: noted

File: ansible/roles/create-vm/tasks/create-ppc64le-vm.yml
Comment (ashwinik30) [reply ansible/roles/create-vm/tasks/create-ppc64le-vm.yml:11]: agree noted

File: ansible/roles/create-vm/tasks/create-ppc64le-vm.yml
Comment (ashwinik30) [reply ansible/roles/create-vm/tasks/create-ppc64le-vm.yml:16]: noted

File: ansible/roles/destroy-vm/tasks/main.yml
Comment (Stringy) [reply ansible/roles/destroy-vm/tasks/main.yml:20]: To be a bit clearer with this one - is the process between the two similar enough that we can do it in the same way for each?

Comment (ashwinik30) [reply ansible/roles/destroy-vm/tasks/main.yml:20]: We cant use same playbook for both because variables are different Also Z creation and destroy  in VPC

Comment (ashwinik30) [reply ansible/roles/destroy-vm/tasks/main.yml:20]: I tried this but it is somehow not applying 
ansible 2.7 supports this apply functionality

File: ansible/roles/provision-vm/tasks/redhat.yml
Comment (ashwinik30) [reply ansible/roles/provision-vm/tasks/redhat.yml:27]: yes  I have made it common for ppc64le and s390 x by adding condition 
when: arch_info == "ppc64le" or arch_info == "s390x"

File: ansible/roles/create-vm/tasks/create-ppc64le-vm.yml
Comment (ashwinik30) [reply ansible/roles/create-vm/tasks/create-ppc64le-vm.yml]: noted done

File: ansible/roles/destroy-vm/tasks/destroy-ppc64le-vm.yml
Comment (ashwinik30) [reply ansible/roles/destroy-vm/tasks/destroy-ppc64le-vm.yml]: not needed , removed

File: .github/workflows/integration-tests-vm-type.yml
Comment (Stringy) [inline .github/workflows/integration-tests-vm-type.yml:104]: I think we can just unconditionally write the key to file (like with s390x above)

File: ansible/ansible.cfg
Comment (Stringy) [inline ansible/ansible.cfg:2]: should be reverted before merge

File: .github/workflows/cpaas.yml
Comment (Stringy) [inline .github/workflows/cpaas.yml:99]: should be reverted before merge

File: ansible/README.md
Comment (Stringy) [inline ansible/README.md:184]: This is duplicating the line above - if we just remove `Z` from line 182, then it covers both cases

File: ansible/roles/destroy-vm/tasks/destroy-ppc64le-vm.yml
Comment (Stringy) [reply ansible/roles/destroy-vm/tasks/destroy-ppc64le-vm.yml:20]: With the changes to VM creation, we can use `inventory_hostname` here which means we don't need to get the VM hostname (lines 2-4) and means we don't need a connection to the host in order to delete the VM

File: ansible/roles/destroy-vm/tasks/main.yml
Comment (Stringy) [reply ansible/roles/destroy-vm/tasks/main.yml:20]: This is fine for now I think. Most important is removing the need to get the VM hostname for deletion

File: ansible/roles/create-vm/vars/ppc64le-vars.yml
Comment (Stringy) [inline ansible/roles/create-vm/vars/ppc64le-vars.yml:1]: bit of a nit-pick but can we name this simply `ppc64le.yml`?

File: .github/workflows/integration-tests-vm-type.yml
Comment (ashwinik30) [reply .github/workflows/integration-tests-vm-type.yml:104]: Will it not overrite the s390x key while we will run for 390x because w are using same file name ?$HOME/.ssh/acs-sshkey_rsa.prv , and  i m guessing it will take latest one

File: ansible/ansible.cfg
Comment (ashwinik30) [reply ansible/ansible.cfg:2]: agree , will revert when we are going to merge it

File: .github/workflows/integration-tests-vm-type.yml
Comment (Stringy) [reply .github/workflows/integration-tests-vm-type.yml:104]: ah yeah I see what you mean - in that case I think this is fine. Ideally we could have different filenames for each key, but we can handle that in the future if needed.

Comment (ashwinik30) [reply .github/workflows/integration-tests-vm-type.yml:104]: yeah ok

File: ansible/README.md
Comment (ashwinik30) [reply ansible/README.md:184]: done

File: ansible/roles/create-vm/vars/ppc64le-vars.yml
Comment (ashwinik30) [reply ansible/roles/create-vm/vars/ppc64le-vars.yml:1]: addressed

File: ansible/group_vars/all.yml
Comment (Stringy) [reply ansible/group_vars/all.yml:43]: can we have a comment here that describes this limitation

File: .github/workflows/integration-tests-vm-type.yml
Comment (Stringy) [reply .github/workflows/integration-tests-vm-type.yml:136]: I think this still stands - We should move this to its own step in the workflow, and only run that step when we're testing a power VM. We also need to make sure it is properly commented, ideally with links to the articles / bug tickets we've used to identify the issue

**Review by Stringy — COMMENTED** [review]
It's looking much better, just a couple of outstanding comments and the stuff that needs to be reverted. After that, it's just waiting for the merge of #1308

File: ansible/group_vars/all.yml
Comment (ashwinik30) [reply ansible/group_vars/all.yml:43]: sure

Comment (ashwinik30) [reply ansible/group_vars/all.yml:43]: done

File: ansible/group_vars/platform_rhel-ppc64le.yml
Comment (ashwinik30) [reply ansible/group_vars/platform_rhel-ppc64le.yml:2]: done

File: .github/workflows/integration-tests-vm-type.yml
Comment (ashwinik30) [reply .github/workflows/integration-tests-vm-type.yml:136]: I did not get actual place where needs to be add this line

Comment (Stringy) [reply .github/workflows/integration-tests-vm-type.yml:136]: Essentially we need:

```
- name: Set MTU for Power VMs
  if: ${{ contains(inputs.vm_type, 'ppc64le') }}
  run: sudo ifconfig eth0 mtu 1000 up
```

Don't forget to include some comments about why it is necessary

File: .github/workflows/integration-tests-vm-type.yml
Comment (Stringy) [inline .github/workflows/integration-tests-vm-type.yml:153]: Can we move this to the script on line 130 instead of added to the create vm step?

File: ansible/README.md
Comment (Stringy) [inline ansible/README.md:22]: [nit] could do with some separation here
```suggestion

To manage IBM Z and Power VMs through IBM Cloud, you will also need to download and install the following ansible collection:
```

File: ansible/README.md
Comment (Stringy) [inline ansible/README.md:25]: [nit] empty line should be removed

File: ansible/group_vars/all.yml
Comment (Stringy) [inline ansible/group_vars/all.yml:41]: [nit] need to be indented in line with the `rhel-ppc64le:` line, and separated across a couple of lines
```suggestion
  # We have taken families name as "p" only. There is limit on chars in instance name 
  # in ibm cloud with 56 chars max
  rhel-ppc64le:
```

File: ansible/group_vars/platform_rhel-ppc64le.yml
Comment (Stringy) [inline ansible/group_vars/platform_rhel-ppc64le.yml:4]: [nit]
```suggestion
# This is workaround in ppc64le vsi creation case as there is issue with ssh connection between ubuntu vm in azure to power vsi
# reference link https://askubuntu.com/questions/1229456/ssh-fails-with-connection-timed-out-in-vpn-and-hangs-here-expecting-ssh2-msg
```

**Review by Stringy — CHANGES_REQUESTED** [review]

File: .github/workflows/integration-tests-vm-type.yml
Comment (mdafsanhossain) [reply .github/workflows/integration-tests-vm-type.yml]: @Stringy Addressed this and the other comments as part of this commit 
https://github.com/stackrox/collector/pull/1296/commits/631d3241a30db882fd8d87be7d4b7e6f9c693259

**Review by Stringy — APPROVED** [review]

**Pull request merged**
